clang/include/clang/AST/BuiltinTypes.def

```diff
@@ -221,6 +225,18 @@ FLOATING_TYPE(Float128, Float128Ty)
 // '__ibm128'
 FLOATING_TYPE(Ibm128, Ibm128Ty)
 
+//===- Decimal floating point types ---------------------------------------===//
+// ISO/IEC TS 18661-2, ISO/IEC TR 24733, and C23 decimal floating-point.
+
+// '_Decimal32'
+DECIMAL_FLOATING_TYPE(DecimalFloat32, DecimalFloat32Ty)
```

Comment:
So the reason why I used `SIGNED_TYPE` in my original set of changes was to simplify lowering and codegen. Also if we are going to encode BID etc then we can treat it as a signed type and we will be passing it to builtins and any operations we would implement would be working with as a signed type for the most part.

I got a lot working this way. I have not really thought through the alternative but it may be the right way to start out. Note sure.

Reply:
`SIGNED_TYPE` and `DECIMAL_FLOATING_TYPE` do the same thing by default (see the macro definitions above). This just allows us to differentiate them when it is advantageous to do so and to be explicit otherwise. We can, and I think should, lower to integer types for builtin/intrinsic operations as you state.